src/FastRouteRouter.php

```diff
@@ -277,41 +255,80 @@ public function generateUri($name, array $substitutions = [], array $options = [
         }
 
         $route   = $this->routes[$name];
-        $path    = $route->getPath();
         $options = ArrayUtils::merge($route->getOptions(), $options);
 
         if (! empty($options['defaults'])) {
             $substitutions = array_merge($options['defaults'], $substitutions);
         }
 
-        foreach ($substitutions as $key => $value) {
-            $pattern = sprintf(
-                '~%s~x',
-                sprintf(self::VARIABLE_REGEX, preg_quote($key))
-            );
-            $path = preg_replace($pattern, $value, $path);
-        }
+        $routeParser        = new RouteParser();
```

Comment:
@xtreamwayz It appears to be stateless, so I think that'd be OK. Maybe do `$routeParser = $this->getRouteParser()` and add a private `getRouteParser()` for lazy instantiation and setting the private property?